plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryType.java

```diff
@@ -277,14 +280,30 @@ private static String quote(String value)
         return "'" + value + "'";
     }
 
-    String convertToString(Object value)
+    String convertToString(Type type, Object value)
     {
+        if (type instanceof DecimalType) {
+            if (isShortDecimal(type)) {
+                return "NUMERIC " + quote(Decimals.toString((long) value, ((DecimalType) type).getScale()));
+            }
+            return "NUMERIC " + quote(Decimals.toString((Slice) value, ((DecimalType) type).getScale()));
+        }
         return toStringConverter.convertToString(value);
     }
 
     public Type getNativeType(BigQueryType.Adaptor typeAdaptor)
     {
         switch (this) {
+            case NUMERIC:
+                Long precision = typeAdaptor.getPrecision();
+                Long scale = typeAdaptor.getScale();
+                if (precision != null && scale != null) {
+                    return createDecimalType(toIntExact(precision), toIntExact(scale));
+                }
+                if (precision != null) {
+                    return createDecimalType(toIntExact(precision));
+                }
+                return createDecimalType(NUMERIC_DATA_TYPE_PRECISION, NUMERIC_DATA_TYPE_SCALE);
```

Comment:
Can we now rename `NUMERIC_DATA_TYPE_PRECISION` and `NUMERIC_DATA_TYPE_SCALE` to express these are defaults?

e.g. `DEFAULT_NUMERIC_TYPE_PRECISION` and `DEFAULT_NUMERIC_TYPE_SCALE`?